config-controller/pkg/client/client.go

```diff
@@ -0,0 +1,303 @@
+package client
+
+//go:generate mockgen-wrapper CachedPolicyClient,PolicyClient
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	storage "github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/size"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var centralHostPort = fmt.Sprintf("central.%s.svc:443", env.Namespace.Setting())
+
+type perRPCCreds struct {
+	svc         v1.AuthServiceClient
+	metadata    map[string]string
+	lastUpdated time.Time
+}
+
+func (c *perRPCCreds) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
+	return c.metadata, nil
+}
+
+func (c *perRPCCreds) RequireTransportSecurity() bool {
+	return true
+}
+
+func (c *perRPCCreds) refreshToken(ctx context.Context) error {
+	getLogger(ctx).Info("Refreshing Central API token")
```

Comment:
nit: debug log instead of info

Reply:
controller logging is trickier than I expected.  There is only info and error in this logging framework, and I found [a controller-runtime issue](https://github.com/kubernetes-sigs/controller-runtime/issues/167#issuecomment-426328544) where one of the maintainers recommended against using verbosity numbers, e.g. (`log.V(2).Info(...)`).  I think there was something mentioned about filtering things out by logger name?? 

Regardless, there is probably a larger discussion on how to properly do logging here, given the different logging framework that is pushed on us by controller-runtime.

Reply:
Created https://issues.redhat.com/browse/ROX-26109 to address this